pkg/server/headscale/server.go

```diff
@@ -72,7 +81,39 @@ func (s *HeadscaleServer) Start() error {
 		}
 	}()
 
-	return <-errChan
+	if s.frps != nil {
```

Comment:
I would recommend the exact thing to be considered as mentioned in the comment above.